### PR TITLE
Fix build warnings

### LINKED ISSUE #3758: Fix build warnings
Build warnings might be caused by broken links, incorrect markup, etc.

### PR DESCRIPTION
**Pull request by andreyaksenov**
- Added the `compat` tutorial to TOC: https://docs.d.tarantool.io/en/doc/fix-warnings/reference/reference_lua/compat/compat_tutorial/
- Fixed numbering in https://docs.d.tarantool.io/en/doc/fix-warnings/book/admin/upgrades/upgrade_standalone/#upgrading-a-standalone-instance.
- Several broken links, for example: https://docs.d.tarantool.io/en/doc/fix-warnings/release/2.10.8/#core.
- Duplication of configuration property: https://www.tarantool.io/en/doc/latest/contributing/docs/api/#contributing-api-docs-confval-example.
- C API - fixed markup for types:
    - https://docs.d.tarantool.io/en/doc/fix-warnings/dev_guide/reference_capi/error/
    - https://docs.d.tarantool.io/en/doc/fix-warnings/dev_guide/reference_capi/fiber/
    - https://docs.d.tarantool.io/en/doc/fix-warnings/dev_guide/reference_capi/tuple/
- SQL - fixed headings:
    - https://docs.d.tarantool.io/en/doc/fix-warnings/reference/reference_sql/sql_statements_and_clauses/#date-part
    - https://docs.d.tarantool.io/en/doc/fix-warnings/reference/reference_sql/sql_statements_and_clauses/#now
- Fixed/removed duplicated IDs.
- Fixed highlighting for several code snippets (incorrect language).
- Removed some files not included in a TOC tree (but still indexed).

### CODE REVIEW

File: doc/contributing/docs/api.rst [hunk 1]
```diff
@@ -179,9 +179,11 @@ Configuration parameters
...
 
-..  literalinclude:: ./_includes/confval_template.rst
+..  literalinclude:: /reference/configuration/cfg_basic.rst
     :language: rst
+    :lines: 219-231
 
 Result:
 
-..  include:: ./_includes/confval_template.rst
+..  literalinclude:: /reference/configuration/cfg_basic.rst
```
Comment (p7nov) [inline doc/contributing/docs/api.rst]: The result looks incorrect:

File: doc/reference/reference_lua/compat.rst [hunk 2]
```diff
@@ -37,22 +37,32 @@ Consider example below:
 
 *   In release 4.0, ``json_esc_slash`` is marked as obsolete, and the old behavior is no longer accessible. Developers are forced to use the new behavior.
 
-Basic compat usage
-------------------
+Basic usage
```
Comment (p7nov) [inline doc/reference/reference_lua/compat.rst:40]: Let's summon @veod32 to check all changes in compat docs.

File: doc/reference/reference_lua/net_box.rst [hunk 3]
```diff
@@ -132,9 +132,7 @@ Below is a list of all ``net.box`` functions.
         *   -   :ref:`conn:eval() <net_box-eval>`                                
             -   Evaluate the expression in a string and execute it                
         *   -   :ref:`conn:call() <net_box-call>`                      
-            -   Call a stored procedure               
-        *   -   :ref:`conn:timeout() <conn-timeout>`
```
Comment (p7nov) [inline doc/reference/reference_lua/net_box.rst:136]: What happened to `conn:timeout()`?

**Review by p7nov — APPROVED** [review]
Thanks for fixing these! A couple of comments from my side.

File: doc/reference/reference_lua/net_box.rst (hunk 3 above)
Comment (andreyaksenov) [reply doc/reference/reference_lua/net_box.rst:136]: I've checked - looks like it was removed from the API

File: doc/contributing/docs/api.rst (hunk 1 above)
Comment (andreyaksenov) [reply doc/contributing/docs/api.rst]: Thanks, fixed:
https://docs.d.tarantool.io/en/doc/fix-warnings/contributing/docs/api/#contributing-api-docs-confval-example

**Review by veod32 — APPROVED** [review]

**Pull request merged**